src/pycell.rs

```diff
@@ -869,6 +922,12 @@ impl<'py, T: PyClass<Frozen = False>> PyRefMut<'py, T> {
             .try_borrow_mut()
             .map(|_| Self { inner: obj.clone() })
     }
+
+    pub(crate) fn downgrade(slf: &Self) -> &PyRef<'py, T> {
+        // `PyRef<T>` and `PyRefMut<T>` have the same layout, and they are behind
+        // a reference so the difference in `Drop` behavior is irrelevant.
+        unsafe { &*(slf as *const Self as *const PyRef<'py, T>) }
```

Comment:
Given we merged #4240, it would be great to use that here and elsewhere in this implementation.

```suggestion
        unsafe { &*ptr_from_ref(slf).cast() }
```